File: Backend/src/main/java/com/cinema/ticketsystem/model/Movie.java
package com.cinema.ticketsystem.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "movies")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Movie {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotBlank
    @Size(max = 200)
    @Column(nullable = false)
    private String title;
    
    @Size(max = 2000)
    @Column(length = 2000)
    private String description;
    
    @NotNull
    @Min(1)
    @Column(nullable = false)
    private Integer durationMinutes;
    
    @NotBlank
    @Size(max = 100)
    @Column(nullable = false)
    private String genre;
    
    @Size(max = 10)
    @Column(length = 10)
    private String rating; // e.g., "PG-13", "R", "G"
    
    @Size(max = 100)
    private String director;
    
    @Size(max = 500)
    private String cast;
    
    @Column(length = 500)
    private String posterUrl;
    
    @Column(length = 500)
    private String trailerUrl;
    
    @NotNull
    @Column(nullable = false)
    private LocalDate releaseDate;
    
    @DecimalMin("0.0")
    @DecimalMax("10.0")
    @Column(precision = 3, scale = 1)
    private BigDecimal averageRating;
    
    @Min(0)
    @Column
    private Integer totalReviews = 0;
    
    @Column(nullable = false)
    private Boolean isActive = true;
    
    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;
    
    @OneToMany(mappedBy = "movie", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Screening> screenings = new ArrayList<>();
    
    @OneToMany(mappedBy = "movie", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Review> reviews = new ArrayList<>();
}

File: cinema-ticket-C#/Model/Movie.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CinemaTicket.Model;

[Table("movies")]
public class Movie
{
    public long Id { get; set; }

    [Required, MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(2000)]
    public string? Description { get; set; }

    [Range(1, int.MaxValue)]
    public int DurationMinutes { get; set; }

    [Required, MaxLength(100)]
    public string Genre { get; set; } = string.Empty;

    [MaxLength(10)]
    public string? Rating { get; set; }

    [MaxLength(100)]
    public string? Director { get; set; }

    [MaxLength(500)]
    public string? Cast { get; set; }

    [MaxLength(500)]
    public string? PosterUrl { get; set; }

    [MaxLength(500)]
    public string? TrailerUrl { get; set; }

    public DateOnly ReleaseDate { get; set; }

    [Range(0, 10)]
    [Column(TypeName = "decimal(3,1)")]
    public decimal? AverageRating { get; set; }

    [Range(0, int.MaxValue)]
    public int TotalReviews { get; set; }

    public bool IsActive { get; set; } = true;

    public ICollection<Screening> Screenings { get; set; } = new List<Screening>();

    [Timestamp]
    public byte[] RowVersion { get; set; } = Array.Empty<byte>();
}
